File: src/main/java/org/jukeboxmc/event/player/PlayerChangeSkinEvent.java
package org.jukeboxmc.event.player;

import org.jukeboxmc.event.Cancellable;
import org.jukeboxmc.player.Player;
import org.jukeboxmc.player.skin.Skin;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerChangeSkinEvent extends PlayerEvent implements Cancellable {

    private Skin skin;

    /**
     * Creates a new {@link PlayerEvent}
     *
     * @param player who represents the player which comes with this event
     */
    public PlayerChangeSkinEvent( Player player, Skin skin ) {
        super( player );
        this.skin = skin;
    }

    public Skin getSkin() {
        return this.skin;
    }

    public void setSkin( Skin skin ) {
        this.skin = skin;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/player/PlayerChangeSkinEvent.kt
package org.jukeboxmc.api.event.player

import org.jukeboxmc.api.event.Cancellable
import org.jukeboxmc.api.player.Player
import org.jukeboxmc.api.player.skin.Skin

class PlayerChangeSkinEvent(
    player: Player,
    private var skin: Skin
) : PlayerEvent(player), Cancellable {

    fun getSkin(): Skin {
        return this.skin
    }

    fun setSkin(skin: Skin) {
        this.skin = skin
    }

}